### ISSUE TITLE
Editing 'Stock and Price Values' clears all property data for each variant

### ISSUE THREAD

**Issue by artistro08** [OFFLINE-GmbH/oc-mall-plugin#149]
When I open a product and I click 'Edit stock and price values' and change the global price, It clears the variant data for each entry. No Idea of why it does this.

Backend Settings for Database is switched to filesystem. 

Flow: 

Open Product (with variants) > Click on 'Edit price and stock values' > Save > 'Refresh' > Variant info gone

**Comment by tobias-kuendig** [comment]
With Variant info you mean the prices for each variant in the table?

**Comment by artistro08** [comment]
I can reproduce this is on https://mall.offline.swiss

The variant info I'm talking about is the things you add in the "Property Groups" section for Categories. The stuff you use for variants (eg. Size, Color, etc). Properties. 

If I had to take a guess of what's going on, When you edit the stock and price values, it'll save the stock and price, but discard the property info.

**Comment by artistro08** [comment]
and for added info, here are all the queries that ran after you edit the Stock and Price Values:

```
select * from `backend_users` where `backend_users`.`id` = '2' limit 1
580μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `backend_user_throttle` where `user_id` = '2' and (`ip_address` = '172.58.83.179' or `ip_address` is null) limit 1
390μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `backend_user_roles` where `backend_user_roles`.`id` is null limit 1
310μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_prices` where `offline_mall_prices`.`priceable_id` is null and `offline_mall_prices`.`priceable_id` is not null and `offline_mall_prices`.`priceable_type` = 'mall.custom_field_option' and price_category_id is null
350μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `system_files` where `system_files`.`attachment_type` = 'mall.custom_field_option' and `system_files`.`attachment_id` is null and `field` = 'image' and `system_files`.`attachment_id` is not null order by `sort_order` asc limit 1
340μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select `name`, `code` from `rainlab_translate_locales` order by `sort_order` asc
370μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_currencies` order by `is_default` desc limit 1
320μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_price_categories` order by `sort_order` asc
380μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_customer_groups` order by `sort_order` asc
330μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_products` where `offline_mall_products`.`id` = '8' and `offline_mall_products`.`deleted_at` is null limit 1
480μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_image_sets` where `offline_mall_image_sets`.`product_id` in ('8')
340μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_product_prices` where variant_id is null and `offline_mall_product_prices`.`product_id` in ('8')
370μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_prices` where `offline_mall_prices`.`priceable_id` in ('8') and `offline_mall_prices`.`priceable_type` = 'mall.product'
400μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_customer_group_prices` where `offline_mall_customer_group_prices`.`priceable_id` in ('8') and `offline_mall_customer_group_prices`.`priceable_type` = 'mall.product'
460μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_product_variants` where `offline_mall_product_variants`.`product_id` in ('8') and `offline_mall_product_variants`.`deleted_at` is null
400μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_products` where `offline_mall_products`.`id` in ('8') and `offline_mall_products`.`deleted_at` is null
480μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_image_sets` where `offline_mall_image_sets`.`id` in ('')
320μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_product_prices` where `offline_mall_product_prices`.`variant_id` in ('43')
330μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_prices` where `offline_mall_prices`.`priceable_id` in ('43') and `offline_mall_prices`.`priceable_type` = 'mall.variant'
330μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_customer_group_prices` where `offline_mall_customer_group_prices`.`priceable_id` in ('43') and `offline_mall_customer_group_prices`.`priceable_type` = 'mall.variant'
310μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_currencies` order by `is_default` desc, `sort_order` asc
410μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `users` where `users`.`id` = '1' limit 1
390μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_customer_groups` where `offline_mall_customer_groups`.`id` in ('') order by `sort_order` asc
270μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
Begin Transaction
/vendor/october/rain/src/Events/Dispatcher.php:197
offlineg_malldemo
Begin Transaction
/vendor/october/rain/src/Database/Connections/Connection.php:59
offlineg_malldemo
select * from `offline_mall_currencies` order by `sort_order` asc
410μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
delete from `offline_mall_prices` where (`priceable_type` = 'mall.product' and `priceable_id` = '8' and `currency_id` = '2') or (`priceable_type` = 'mall.variant' and `priceable_id` = '43' and `currency_id` = '2') or (`priceable_type` = 'mall.product' and `priceable_id` = '8' and `currency_id` = '1') or (`priceable_type` = 'mall.variant' and `priceable_id` = '43' and `currency_id` = '1') or (`priceable_type` = 'mall.product' and `priceable_id` = '8' and `currency_id` = '3') or (`priceable_type` = 'mall.variant' and `priceable_id` = '43' and `currency_id` = '3')
660μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
delete from `offline_mall_customer_group_prices` where (`priceable_type` = 'mall.product' and `priceable_id` = '8' and `currency_id` = '2') or (`priceable_type` = 'mall.variant' and `priceable_id` = '43' and `currency_id` = '2') or (`priceable_type` = 'mall.product' and `priceable_id` = '8' and `currency_id` = '1') or (`priceable_type` = 'mall.variant' and `priceable_id` = '43' and `currency_id` = '1') or (`priceable_type` = 'mall.product' and `priceable_id` = '8' and `currency_id` = '3') or (`priceable_type` = 'mall.variant' and `priceable_id` = '43' and `currency_id` = '3')
400μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select count(*) as aggregate from `offline_mall_products` where `id` <> '8' and `slug` = 'variant-test-1' and `offline_mall_products`.`deleted_at` is null
490μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
update `offline_mall_product_prices` set `price` = '2000', `updated_at` = '2019-02-28 12:11:20' where `id` = '25'
390μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_product_prices` where variant_id is null and `offline_mall_product_prices`.`product_id` in ('8')
440μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_property_values` where `offline_mall_property_values`.`product_id` in ('8')
390μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_properties` where `offline_mall_properties`.`id` in ('3', '8') and `offline_mall_properties`.`deleted_at` is null
370μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_prices` where `offline_mall_prices`.`priceable_id` in ('8') and `offline_mall_prices`.`priceable_type` = 'mall.product'
430μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_product_prices` where `offline_mall_product_prices`.`variant_id` in ('43')
380μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_prices` where `offline_mall_prices`.`priceable_id` in ('43') and `offline_mall_prices`.`priceable_type` = 'mall.variant'
330μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_products` where `offline_mall_products`.`deleted_at` is null limit 1
400μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_image_sets` where `offline_mall_image_sets`.`product_id` in ('1')
390μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `system_files` where `system_files`.`attachment_id` in ('1') and `system_files`.`attachment_type` = 'mall.imageset' and `field` = 'images' order by `sort_order` asc
500μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_product_prices` where variant_id is null and `offline_mall_product_prices`.`product_id` in ('1')
410μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_property_values` where ((`product_id` = '1' and `variant_id` is null)) and `offline_mall_property_values`.`variant_id` in ('43')
300μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_prices` where `offline_mall_prices`.`priceable_id` = '8' and `offline_mall_prices`.`priceable_id` is not null and `offline_mall_prices`.`priceable_type` = 'mall.product'
420μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select `offline_mall_categories`.*, `offline_mall_category_product`.`product_id` as `pivot_product_id`, `offline_mall_category_product`.`category_id` as `pivot_category_id`, `offline_mall_category_product`.`sort_order` as `pivot_sort_order` from `offline_mall_categories` inner join `offline_mall_category_product` on `offline_mall_categories`.`id` = `offline_mall_category_product`.`category_id` where `offline_mall_category_product`.`product_id` = '8' and `offline_mall_categories`.`deleted_at` is null order by `nest_left` asc
920μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_currencies` where `offline_mall_currencies`.`id` = '2' order by `sort_order` asc limit 1
380μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_brands` where `offline_mall_brands`.`id` is null order by `sort_order` asc limit 1
330μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_index` where (`index` = 'products' and `product_id` = '8' and `variant_id` is null and `is_ghost` = '0') limit 1
690μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
update `offline_mall_index` set `on_sale` = '0', `property_values` = '{\"8\":[\"s\"],\"3\":[\"black\"]}', `prices` = '{\"EUR\":2000}' where `id` = '51'
350μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
delete from `offline_mall_index` where `index` = 'variants' and `product_id` = '8'
360μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_property_values` where `offline_mall_property_values`.`product_id` = '8' and `offline_mall_property_values`.`product_id` is not null
480μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_index` where (`index` = 'variants' and `product_id` = '8' and `variant_id` = '43' and `is_ghost` = '0') limit 1
440μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
insert into `offline_mall_index` (`index`, `product_id`, `variant_id`, `is_ghost`, `brand`, `stock`, `sales_count`, `on_sale`, `published`, `category_id`, `property_values`, `sort_orders`, `prices`, `parent_prices`, `customer_group_prices`, `created_at`) values ('variants', '8', '43', '0', '', '10', '0', '0', '0', '[5]', '{\"8\":[\"s\"],\"3\":[\"black\"]}', '[]', '{\"EUR\":2000}', '{\"EUR\":2000}', '[]', '2019-02-28 12:04:59')
550μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_product_variants` where `offline_mall_product_variants`.`id` is null and `offline_mall_product_variants`.`deleted_at` is null limit 1
270μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
update `offline_mall_products` set `updated_at` = '2019-02-28 12:11:20' where `offline_mall_products`.`id` = '8'
640μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
update `offline_mall_product_variants` set `updated_at` = '2019-02-28 12:11:20' where `offline_mall_product_variants`.`id` is null
210μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_product_variants` where `offline_mall_product_variants`.`id` = '43' and `offline_mall_product_variants`.`deleted_at` is null limit 1
350μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_products` where `offline_mall_products`.`id` in ('8') and `offline_mall_products`.`deleted_at` is null
450μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
delete from `offline_mall_property_values` where `variant_id` = '43'
350μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_property_values` where `variant_id` = '43'
440μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_products` where `offline_mall_products`.`id` = '8' and `offline_mall_products`.`deleted_at` is null limit 1
500μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select count(*) as aggregate from `offline_mall_products` where `id` <> '8' and `slug` = 'variant-test-1' and `offline_mall_products`.`deleted_at` is null
430μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
delete from `offline_mall_property_values` where `variant_id` = '43'
330μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_property_values` where `variant_id` = '43'
550μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
delete from `offline_mall_property_values` where `variant_id` = '43'
450μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_property_values` where `variant_id` = '43'
340μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
Commit Transaction
/vendor/october/rain/src/Events/Dispatcher.php:197
offlineg_malldemo
Commit Transaction
/vendor/october/rain/src/Database/Connections/Connection.php:59
offlineg_malldemo
select * from `offline_mall_property_values` where `offline_mall_property_values`.`product_id` in ('8')
470μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_product_variants` where `offline_mall_product_variants`.`product_id` in ('8') and `offline_mall_product_variants`.`deleted_at` is null
440μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_products` where `offline_mall_products`.`deleted_at` is null limit 1
530μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_property_values` where ((`product_id` = '1' and `variant_id` is null)) and `offline_mall_property_values`.`variant_id` in ('43')
290μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_index` where (`index` = 'products' and `product_id` = '8' and `variant_id` is null and `is_ghost` = '0') limit 1
780μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
update `offline_mall_index` set `property_values` = '[]', `created_at` = '2019-02-28 12:03:21' where `id` = '51'
5.46ms
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
delete from `offline_mall_index` where `index` = 'variants' and `product_id` = '8'
5.99ms
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_property_values` where `offline_mall_property_values`.`product_id` = '8' and `offline_mall_property_values`.`product_id` is not null
500μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
select * from `offline_mall_index` where (`index` = 'variants' and `product_id` = '8' and `variant_id` = '43' and `is_ghost` = '0') limit 1
460μs
/vendor/october/rain/src/Database/Connections/Connection.php:44
offlineg_malldemo
insert into `offline_mall_index` (`index`, `product_id`, `variant_id`, `is_ghost`, `brand`, `stock`, `sales_count`, `on_sale`, `published`, `category_id`, `property_values`, `sort_orders`, `prices`, `parent_prices`, `customer_group_prices`, `created_at`) values ('variants', '8', '43', '0', '', '10', '0', '0', '0', '[5]', '[]', '[]', '{\"EUR\":2000}', '{\"EUR\":2000}', '[]', '2019-02-28 12:04:59')
```

**Comment by tobias-kuendig** [comment]
I was able to reproduce the issue, thanks @artistro08. I'll look into it.

**Comment by tobias-kuendig** [comment]
A fix has been released as v1.1.10. Please report if the issue isn't fixed for you.

Thanks again for the help, @artistro08! :+1:

**Comment by artistro08** [comment]
You're amazing! Thank you 
![bitmoji](https://render.bitstrips.com/v2/cpanel/f0616061-a98b-4deb-8f3b-9c8c2f3f6655-59e8a5fc-4cb3-4c83-a910-a2ec496da89b-v1.png?transparent=1&palette=1&width=246)
